tools/clang/test/CodeGenDXIL/hlsl/objects/NodeObjects/mesh-node-input-parameters.hlsl

```diff
@@ -5,16 +5,17 @@
 // Test all valid mesh node input parameters work
 
 // CHECK: define void @node01()
-// CHECK:  %[[tid_x:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 0)  ; ThreadId(component)
-// CHECK:  %[[tid_y:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 1)  ; ThreadId(component)
-// CHECK:  %[[tid_z:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 2)  ; ThreadId(component)
-
-// CHECK:  %[[ftid:.+]] = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)  ; FlattenedThreadIdInGroup()
 
 // CHECK:  %[[tid_group_x:.+]] = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 0)  ; ThreadIdInGroup(component)
 // CHECK:  %[[tid_group_y:.+]] = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 1)  ; ThreadIdInGroup(component)
 // CHECK:  %[[tid_group_z:.+]] = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 2)  ; ThreadIdInGroup(component)
 
+// CHECK:  %[[ftid:.+]] = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)  ; FlattenedThreadIdInGroup()
+
+// CHECK:  %[[tid_x:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 0)  ; ThreadId(component)
+// CHECK:  %[[tid_y:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 1)  ; ThreadId(component)
+// CHECK:  %[[tid_z:.+]] = call i32 @dx.op.threadId.i32(i32 93, i32 2)  ; ThreadId(component)
+
```

Comment:
Added checks for groupId, but I am not sure why this reordering happened.

Reply:
I wish I knew.  I wonder if it's the ordering of this `GenerateDxilComputeAndNodeCommonInputs` call after the other calls for mesh node specifically:
https://github.com/microsoft/DirectXShaderCompiler/pull/6482/files#diff-9406e37b16c71f4146d9c673b5e6dc8408024b2411bdd3212095df9c375d7907R1817

Can we call `GenerateDxilComputeAndNodeCommonInputs` before the `if (props.IsMeshNode()) {` branch?  Does it make a difference to the ordering of the output instructions for some reason?

Reply:
Yes, it does result in a significant difference:
```
%1 = load %dx.types.Handle, %dx.types.Handle* @"\01?outbuf@@3V?$RWStructuredBuffer@V?$vector@I$02@@@@A", align 4
  %2 = call i32 @dx.op.threadId.i32(i32 93, i32 0)  ; ThreadId(component)
  %3 = call i32 @dx.op.threadId.i32(i32 93, i32 1)  ; ThreadId(component)
  %4 = call i32 @dx.op.threadId.i32(i32 93, i32 2)  ; ThreadId(component)
  %5 = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)  ; FlattenedThreadIdInGroup()
  %6 = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 0)  ; ThreadIdInGroup(component)
  %7 = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 1)  ; ThreadIdInGroup(component)
  %8 = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 2)  ; ThreadIdInGroup(component)
```
The ordering is restored, but now theadIdInGroup comes AFTER instead of BEFORE. So now there's another reordering issue!
I'll leave things as they are.